### ISSUE TITLE
BUG: After importing fiona (version >= 1.8.18), any request performed with the requests library using a self-signed certificate leads to an SSL error

### ISSUE THREAD

**Issue by larsderidder** [Toblerity/Fiona#1091]
## Expected behavior and actual behavior.

When running the below code, providing a valid API key on the command line, I get the below error. I expect a normal JSON request (which is indeed what I get when either not importing fiona, or downgrading to 1.8.17 or lower).

```
❯ python main.py <REDACTED>
Traceback (most recent call last):
  File "/home/lars/.virtualenvs/geopandas-issue/lib/python3.9/site-packages/urllib3/connectionpool.py", line 703, in urlopen
    httplib_response = self._make_request(
  File "/home/lars/.virtualenvs/geopandas-issue/lib/python3.9/site-packages/urllib3/connectionpool.py", line 386, in _make_request
    self._validate_conn(conn)
  File "/home/lars/.virtualenvs/geopandas-issue/lib/python3.9/site-packages/urllib3/connectionpool.py", line 1040, in _validate_conn
    conn.connect()
  File "/home/lars/.virtualenvs/geopandas-issue/lib/python3.9/site-packages/urllib3/connection.py", line 414, in connect
    self.sock = ssl_wrap_socket(
  File "/home/lars/.virtualenvs/geopandas-issue/lib/python3.9/site-packages/urllib3/util/ssl_.py", line 449, in ssl_wrap_socket
    ssl_sock = _ssl_wrap_socket_impl(
  File "/home/lars/.virtualenvs/geopandas-issue/lib/python3.9/site-packages/urllib3/util/ssl_.py", line 493, in _ssl_wrap_socket_impl
    return ssl_context.wrap_socket(sock, server_hostname=server_hostname)
  File "/usr/lib/python3.9/ssl.py", line 500, in wrap_socket
    return self.sslsocket_class._create(
  File "/usr/lib/python3.9/ssl.py", line 1040, in _create
    self.do_handshake()
  File "/usr/lib/python3.9/ssl.py", line 1309, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:1129)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/lars/.virtualenvs/geopandas-issue/lib/python3.9/site-packages/requests/adapters.py", line 440, in send
    resp = conn.urlopen(
  File "/home/lars/.virtualenvs/geopandas-issue/lib/python3.9/site-packages/urllib3/connectionpool.py", line 785, in urlopen
    retries = retries.increment(
  File "/home/lars/.virtualenvs/geopandas-issue/lib/python3.9/site-packages/urllib3/util/retry.py", line 592, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='api.kvk.nl', port=443): Max retries exceeded with url: /api/v1/zoeken?straatnaam=Eemsgolaan&plaats=Groningen&pagina=1&aantal=1&type=hoofdvestiging%2Cnevenvestiging (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:1129)')))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/lars/workspace/geopandas-issue/main.py", line 31, in <module>
    do_search(sys.argv[1])
  File "/home/lars/workspace/geopandas-issue/main.py", line 26, in do_search
    response = session.get(URL_ZOEKEN, params=params)
  File "/home/lars/.virtualenvs/geopandas-issue/lib/python3.9/site-packages/requests/sessions.py", line 542, in get
    return self.request('GET', url, **kwargs)
  File "/home/lars/.virtualenvs/geopandas-issue/lib/python3.9/site-packages/requests/sessions.py", line 529, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/lars/.virtualenvs/geopandas-issue/lib/python3.9/site-packages/requests/sessions.py", line 645, in send
    r = adapter.send(request, **kwargs)
  File "/home/lars/.virtualenvs/geopandas-issue/lib/python3.9/site-packages/requests/adapters.py", line 517, in send
    raise SSLError(e, request=request)
requests.exceptions.SSLError: HTTPSConnectionPool(host='api.kvk.nl', port=443): Max retries exceeded with url: /api/v1/zoeken?straatnaam=Eemsgolaan&plaats=Groningen&pagina=1&aantal=1&type=hoofdvestiging%2Cnevenvestiging (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:1129)')))
```

## Steps to reproduce the problem.

To run the below code, a API key for the specific API, plus its certificates are required. The API key I unfortunately cannot provide as it is a paid service. The required certificates however are found here: https://developers.kvk.nl/support/tls-certificate-chain-trust-instructions.

I do not mind rewriting this example to use a different API, but I couldn't find an open API with self-signed certificates (I suppose that is quite rare).

```python
import sys
import os
import requests
# Commenting out this line fixes the issue
import fiona

URL_ZOEKEN = "https://api.kvk.nl/api/v1/zoeken"

CERT_FILE_NAME = "Staat_der_Nederlanden_Private_Root_CA_-_G1.crt"
"""Should be in the same folder as this file."""


def do_search(api_key):
    session: requests.Session = requests.Session()
    session.headers.update({"apikey": api_key})
    session.verify = os.path.join(
        os.path.abspath(os.path.dirname(__file__)), CERT_FILE_NAME
    )

    params = dict(
        straatnaam="Eemsgolaan",
        plaats="Groningen",
        pagina=1,
        aantal=1,
        type="hoofdvestiging,nevenvestiging",
    )
    response = session.get(URL_ZOEKEN, params=params)
    print(response.text)


if __name__ == '__main__':
    do_search(sys.argv[1])
```

The only two dependencies installed through pip are `requests` and `fiona`.

## Operating system

I have verified this issue on Manjaro Linux, but first encountered it within a Docker image built on top of python:3.9-slim (which is Debian based I believe).

## Fiona and GDAL version and provenance

Any version of Fiona >= 1.8.18 installed from PyPI. My pip version is 22.0.3 currently.

I don't know what is meant with GDAL version and provenance.

**Comment by snowman2** [comment]
https://github.com/OSGeo/gdal/pull/3025

It might be a good idea to use `GDAL_CURL_CA_BUNDLE`

**Comment by sgillies** [comment]
Fixed in 1.9a1.
